api/hotels/hotels.controller.js

```diff
@@ -2,23 +2,27 @@ const mongoose = require('mongoose');
 const Hotels = mongoose.model('hotels');
 const boom = require('boom');
 
-// exports.createHotel = (req, res, next) => {
-
-//     let NewHotel = new Hotels(req.body);
-
-//     NewHotel.save ((err, hotel) => {
-//         if (err) {
-//            next(boom.unauthorized(err.toString()));
-//         }
-
-//         else {
-//             res.json(hotel);
-
-//         }
-//     })};
-
-
-exports.search =(req, res, next) => {
-
-
-}
\ No newline at end of file
+exports.search = (req, res, next) => {
+    var search = req.query.data;
+
+    if (search.length <= 2) {
+        next(boom.badImplementation("Please provide atleast 3 letters to search"));
+    } else {
+        Hotels.find({
+                name: {
+                    $regex: search,
+                    $options: 'i'
+                }
+            })
+            .then((data) => {
+                if (data.length === 0) {
+                    next(boom.forbidden('No such hotels found with this keyword found'));
+                } else {
+                    res.send(data)
```

Comment:
remove else. Instead use this

if(data.length == 0){
return next(boom.forbidden('No such hotels found with this keyword found'));
}